### PR TITLE
Select locale based on query param

### PR DESCRIPTION
**Pull request by metamben**
See https://github.com/metabase/metabase/issues/22561 for context.

### CODE REVIEW

File: test/metabase/server/routes/index_test.clj [hunk 1]
```diff
@@ -45,5 +45,32 @@
...
+  (testing "a valid override is honored no matter what the user locale is"
+    (is (= {"charset"      "utf-8"
+            "headers"      {"mime-version"              "1.0"
+                            "content-type"              "text/plain; charset=UTF-8"
+                            "content-transfer-encoding" "8bit"
+                            "x-generator"               "POEditor.com"
+                            "project-id-version"        "Metabase"
+                            "language"                  "es"
+                            "plural-forms"              "nplurals=2; plural=(n != 1);"}
+            "translations" {"" {"Your database has been added!" {"msgstr" ["¡Tu base de datos ha sido añadida!"]}}}}
```
Comment (adam-james-v) [inline test/metabase/server/routes/index_test.clj:61]: Probably not likely, but I think if a translator ever revises this particular translation, the test will fail, I think.

File: test/metabase/server/routes/index_test.clj [hunk 2]
```diff
@@ -45,5 +45,32 @@
...
+           (some->
+            (binding [i18n/*user-locale* "xx"]
+              (#'index/load-localization "es"))
+            json/parse-string
+            (update "translations" select-keys [""])
+            (update-in ["translations" ""] select-keys ["Your database has been added!"])))))
+
+  (testing "an invalid override causes a fallback to English"
+    (is (= {"headers"      {"language" "yy", "plural-forms" "nplurals=2; plural=(n != 1);"}
```
Comment (adam-james-v) [inline test/metabase/server/routes/index_test.clj:70]: The header has "language" "yy"... if we're actually defaulting to English, does it make sense to send "yy"? 

I think this was talked about before... perhaps it's to let the frontend know what language code failed? Not a blocker, I don't think.

**Review by adam-james-v — APPROVED** [review]

File: test/metabase/server/routes/index_test.clj (hunk 2 above)
Comment (metamben) [reply test/metabase/server/routes/index_test.clj:70]: This [conversation](https://metaboat.slack.com/archives/CKZEMT1MJ/p1651173809142869) is exactly the thing why I asked specifically you.
Note that my new tests are in line with the existing ones. I think the logic is, that we are not "correcting" the frontend, we just give it the best translations we can.

Comment (adam-james-v) [reply test/metabase/server/routes/index_test.clj:70]: Yea, that sounds right. Sorry, it was a long thread and that detail got fuzzy in my mind. I'm happy with it :)

**Pull request merged**